generic/MaplGeneric.F90

```diff
@@ -2,7 +2,6 @@ module MaplGeneric
    use mapl_AbstractComponent
    use mapl_MaplComponent
    use mapl_ComponentSpecification
-   use mapl_VariableSpecification
```

Comment:
Is this change necessary?   The other layers of MAPL don't yet use this (new) data type, but they will eventually.

Reply:
@tclune This change attempts to fix the following compiler error:
```
[...]/GOCART/ESMF/GOCART2G_GridComp/CA2G_GridComp/CA2G_GridCompMod.F90(894): error #6405: The same named entity from different modules and/or program units cannot be referenced.   [MAPL_VARSPEC]
    type(MAPL_VarSpec), pointer       :: InternalSpec(:)
```

Module `mapl_VariableSpecification` exposes derived type `MAPL_VarSpec`, which is also exposed by module `MAPL_VarSpecMod`. Based on last week's discussions, my understanding is that `MAPL_VarSpecMod` is the proper "owner" of `MAPL_VarSpec`.

Reply:
I thought the solution we discussed was a change in MAPL_Generic.F90.   `MAPL/generic` is the proper owner of this module and should be exporting this.

The error you refer to should not happen when starting a clean build.    If it does, lets handle this as a separate hotfix on main.

(These changes otherwise all need to be redone from a base off of develop.  Sorry)